hadoop-ozone/cli-admin/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/VolumeSubCommand.java

```diff
@@ -0,0 +1,222 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hdds.scm.cli.datanode;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos.VolumeInfoProto;
+import org.apache.hadoop.hdds.protocol.proto.StorageContainerLocationProtocolProtos.GetVolumeInfosResponseProto;
+import org.apache.hadoop.hdds.scm.cli.ScmSubcommand;
+import org.apache.hadoop.hdds.scm.client.ScmClient;
+import org.apache.hadoop.hdds.scm.datanode.VolumeInfo;
+import org.apache.hadoop.hdds.server.JsonUtils;
+import org.apache.hadoop.ozone.utils.FormattingCLIUtils;
+import org.apache.hadoop.util.StringUtils;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * We provide a set of volume commands to display the disk information of the DataNode.
+ */
+@Command(
+    name = "volumes",
+    description = "Display the list of volumes on the DataNode.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class VolumeSubCommand extends ScmSubcommand {
+
+  // Display it in JSON format.
+  @Option(names = { "--json" },
+       defaultValue = "false",
+       description = "Format output as JSON.")
+  private boolean json;
+
+  // Display it in TABLE format.
+  @Option(names = { "--table" },
+       defaultValue = "false",
+       description = "Format output as Table.")
+  private boolean table;
+
+  // PageSize refers to the number of items displayed per page
+  // in a paginated view.
+  @Option(names = { "--pageSize" },
+       defaultValue = "20",
+       description = "The number of volume information items displayed per page.")
+  private int pageSize;
+
+  // The current page.
+  @Option(names = { "--currentPage" },
+       defaultValue = "1",
+       description = "The current page.")
+  private int currentPage;
+
+  /**
+   * We have designed a new option called 'show',
+   * which includes two selectable configurations:
+   * 'failed' is used to display failed disks,
+   * and 'normal' is used to display normal disks.
+   */
+  @Option(names = { "--displayMode" },
+      defaultValue = "all",
+      description = "failed is used to display failed disks, " +
+      "normal is used to display normal disks.")
+  private String displayMode;
+
+  // The UUID identifier of the DataNode.
+  @Option(names = { "--uuid" },
+      defaultValue = "",
+      description = "failed is used to display failed disks, " +
+      "normal is used to display normal disks.")
```

Comment:
description seems to be dup of `--displayMode`.  (Same for `--hostName`.)

Reply:
I will improve the help information for `displayMode`, `hostname`, and `uuid`.